internal/attrs/attrs.go

```diff
@@ -147,3 +150,25 @@ func (a *Attributes) AddToOTG(top gosnappi.Config, ap *ondatra.Port, peer *Attri
 		ip.SetAddress(a.IPv6).SetGateway(peer.IPv6).SetPrefix(int32(a.IPv6Len))
 	}
 }
+
+// AddToDut attaches a subinterface to a network instance
+func (a *Attributes) SetOnDUT(t *testing.T, dconf *ondatra.Config, i *telemetry.Interface) {
```

Comment:
I still have a few concerns with this idiom.

Originally, the use case I intended was that attrs.AddToDUT would be a complete solution for configuring an interface: take the Attributes, generate *telemetry.Interface, replace the interface on dut, and optionally add the interface to the NI. I want something functionally equivalent to configInterfaceDUT as a one stop shop for configuring the interface on the DUT and add the interface to the NI when needed.

But looking a little deeper, I realized that static_arp_test (and potentially others) still need to populate more interface leaves than what attrs has, and this is why they don't use attrs.NewInterface() but instead built their own *telemetry.Interface. It will be hard to incorporate test-specific leaves without adding unnecessary complexity to attrs, unless we refactor the tests to add the test-specific leaves separately after an interface has been configured with the common leaves.

It seems that applying the explicit interface in VRF deviation transparently to tests will require some refactoring of existing tests, so is that the reason you opted to simplify the problem by only adding the subinterface to the network instance?

If that's the approach you want to take, then I have a few more suggestions.

 1. Make this a standalone function, not a method of Attributes. Name it `AssignToNetworkInstance` so it more accurately reflects the functionality provided.
 2. It should take *ondatra.DUTDevice as argument, not *ondatra.Config.
 3. It should take the network instance name as an argument.
 4. It should take the interface name as a string, not *telemetry.Interface.
 5. It should take the subinterface index as an argument rather than hard-coded to 0.

Tests that want to put an interface to a non-default NI should be able to use this function as well, so at least for non-default NI, the behavior should not be conditional on deviations.ExplicitInterfaceInVRF. It would be more accurate to rename the deviation to deviations.ExplicitInterfaceInDefaultVRF so it is clear that an interface will always be put into a non-default VRF, but may be no-op if putting it in default VRF is optional.

Reply:
> It seems that applying the explicit interface in VRF deviation transparently to tests will require some refactoring of existing tests, so is that the reason you opted to simplify the problem by only adding the subinterface to the network instance?

The original idea for this PR/Deviation was simply to assign an (sub)interface to a Network Instance. It seems like maybe the idea got muddled in various conversations. I think having a complete solution to configure the interface might be a little out of scope for this PR, perhaps something can be raised at a later time and the community can contribute.

> * Make this a standalone function, not a method of Attributes. Name it `AssignToNetworkInstance` so it more accurately reflects the functionality provided.
> * It should take *ondatra.DUTDevice as argument, not *ondatra.Config.
> * It should take the network instance name as an argument.
> * It should take the interface name as a string, not *telemetry.Interface.
> * It should take the subinterface index as an argument rather than hard-coded to 0.

With regards to your first point, which file should contain this function?
I previously had `si` as a uint32  argument passed to the function but you had mentioned that it would be safe to always assume `0` as the subinterface. Is this no longer the case and you'd rather I revert back to what I initially coded?

Once I have those clarifications, I'll go ahead and make the changes.